### ISSUE TITLE
DropDown support in the web page when doing Automation tasks

### ISSUE THREAD

**Issue by Shiv270895** [lavague-ai/LaVague#417]
**Is your feature request related to a problem? Please describe.**
I was creating a automation agent using LaVague but I was not able to create one as when in the dropdown section the action engine was not able to follow the instruction and it went on a loop.

**Describe the solution you'd like**
It should follow the instruction for the dropdown as well because in many web pages we can see the dropdowns.

**Describe alternatives you've considered**
The alternatives which i have considered was through skipping those dropdown and see weather the agent is working as expected or not.

**Additional context**
If this feature is already there then please provide me with the link so that i can go through it and complete my agent. I checked the documentation and the example's which are present but couldn't find anything relatable.

**Comment by dhuynh95** [comment]
Will have a look.

I think we should emphasize using dropdowns to the world model that tends to ignore it.

**Comment by dhuynh95** [comment]
Opened PR: https://github.com/lavague-ai/LaVague/pull/418

**Comment by Shiv270895** [comment]
Okay Thankyou, Will test it from my end as well.

**Comment by Shiv270895** [comment]
Hello @dhuynh95 
I tried the dropdowns but still i am not able to achieve my objective.

Below is the Instruction which I am giving before execution
- Job Title is dropdown list. Do not take other actions until the dropdown is Opened.
- Click on IT Manager in the Job Title dropdown list
- Do not take other actions until the dropdown is Closed.

This is the error which I am getting while executing my script
- Next engine: Navigation Engine
- Instruction: Click on the 'Job Title' dropdown list.
- 2024-07-18 09:55:51,147 - ERROR - Navigation error: No element found.

Please if you can help me in knowing where i am going wrong or any alternate approach which i should take to accomplish my objective. 
It would be really helpful.

Thanks & Regards,
Shiv kumar

**Comment by Shiv270895** [comment]
Hi @lyie28 ,

Sure I am sharing the full code with you, Please have a look.

`!pip install lavague llama-index-multi-modal-llms-gemini llama-index-embeddings-gemini llama-index-llms-gemini llama-index-embeddings-huggingface llama-index-llms-mistralai -q`

`from llama_index.embeddings.gemini import GeminiEmbedding
from llama_index.llms.gemini import Gemini
from llama_index.multi_modal_llms.gemini import GeminiMultiModal
from llama_index.llms.mistralai import MistralAI
from lavague.core import WorldModel, ActionEngine
from lavague.core.agents import WebAgent
from lavague.core.context import Context
from lavague.drivers.selenium import SeleniumDriver
from llama_index.embeddings.huggingface import HuggingFaceEmbedding
from llama_index.core import Settings



from google.colab import userdata
api_key = userdata.get('google_api_key')

llm = Gemini(model_name="models/gemini-1.5-flash", api_key=api_key)
mm_llm =  GeminiMultiModal(model_name="models/gemini-1.5-flash", api_key=api_key)
Settings.embed_model = HuggingFaceEmbedding(
    model_name="BAAI/bge-small-en-v1.5"
)
embedding = Settings.embed_model

context = Context(llm, mm_llm, embedding)

# Initialize context with our custom elements
context = Context(llm, mm_llm, embedding)

# Initialize the Selenium driver
selenium_driver = SeleniumDriver()

# Initialize a WorldModel and ActionEnginem passing them the custom context
world_model = WorldModel.from_context(context)
action_engine = ActionEngine.from_context(context, selenium_driver)

# Create your agent
agent = WebAgent(world_model, action_engine)

agent.get("https://docs.google.com/forms/d/e/1FAIpQLSdv5N6nzvmTSu2aJ-B1Li2DWgiC5VICy2keWP28gaR0qM3hZg/viewform?usp=pp_url")


instruction = """
Fill out this Contact Us form
- Name is Shiv kumar
- Email is abcd@xyz.com
- Phone number is +918464862836
- How can we help you is a dropdown list. Do not take other actions until the dropdown is Opened.
- Click on Support in the dropdown list.
- Do not take other actions until the dropdown is Closed.
- Enter Message 'Hi How are you'
- Click on Submit.
"""

agent.demo(instruction)
`

Here I am using Gemini model and Hugging face model for embeddings. 

Please have a look and tell what modification i can make so that the dropdown starts to function.

Thanks & Regards
Shiv Kumar

**Comment by lyie28** [comment]
Hi @Shiv270895 - apologies, I missed your reply.

We added some code to handle drop-downs around a couple of weeks back which seem to solve your issue.

I just tested this now with the latest packages and it worked well BUT with a couple of changes. 

1. I used `agent.run(objective=instruction, display=True)` because there seemed to be some issue with `demo()` which I am going to flag & debug now 🔎
2. It did not perform too well with `models/gemini-1.5-flash` as the `mm_llm` as you have in your code, but when I upgrade to `gemini-1.5-pro` it was able to complete the task.

https://github.com/user-attachments/assets/69b5daf8-1acd-43a4-8610-0a26e89311a7
